security/pkg/nodeagent/sds/monitoring.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sds
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	totalPushCounts = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "citadel_agent",
+		Subsystem: "sds_service",
+		Name:      "total_push_count",
+		Help:      "The total number of SDS pushes.",
+	})
+
+	pendingPushPerConnCounts = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "citadel_agent",
+		Subsystem: "sds_service",
+		Name:      "push_count",
+		Help:      "The number of secret pushes to an active SDS connection.",
+	}, []string{"resourcePerConn"})
+
+	pushPerConnCounts = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "citadel_agent",
+		Subsystem: "sds_service",
+		Name:      "push_count",
+		Help:      "The number of secret pushes to an active SDS connection.",
+	}, []string{"resourcePerConn"})
+
+	pushFailurePerConnCounts = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "citadel_agent",
+		Subsystem: "sds_service",
+		Name:      "push_failure_count",
```

Comment:
in pilot it is `pilot_xds_push_errors` so maybe `push_errors` to be a bit more consistent? Also I think in general `_count` is not added